### PR TITLE
Working comments

### PR DESCRIPTION
**Pull request by mam249**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/data/Task.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Copyright 2019 Google LLC
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/data/Task.java]: 2020

File: portfolio/src/main/java/com/google/sps/data/Task.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    private final String displayName;
+    private final long timestamp;
+
+    public Task(long id, String comment, String displayName, long timestamp) {
+        this.id = id;
+        this.comment = comment;
+        this.displayName = displayName;
+        this.timestamp = timestamp;
+    }
+}
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/data/Task.java]: nit: newline

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -32,15 +32,23 @@ function addRandomImage() {
...
+  });
 }
 
-async function getRandomMessageUsingAsyncAwait() {
-  const response = await fetch('/data');
-  const quote = await response.text();
-  document.getElementById('message-container').innerText = quote;
-}
\ No newline at end of file
+/** Creates an element that represents a task,*/
```
Comment (lminchillo) [inline portfolio/src/main/webapp/script.js]: nit: remove `,` at the end of the comment

File: portfolio/src/main/java/com/google/sps/data/Task.java [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data;
+
+
+public final class Task {
+
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/data/Task.java:23]: nit: remove newline.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 5]
```diff
@@ -28,24 +36,60 @@
 
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: nit: remove TODO now that comments data is being handled. Instead, add comment describing what the servlet does.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 6]
```diff
@@ -28,24 +36,60 @@
...
   @Override
-  public void init() {
-    trial_array = new ArrayList<>();
-    trial_array.add("First hard-coded message");
-    trial_array.add("Second hard-coded message");
-    trial_array.add("Third hard-coded message");
-    json_array = new Gson().toJson(trial_array);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment-box");
+    String displayName = request.getParameter("display-name");
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: We can avoid accessing this parameter if anonymous is set to true. I'd move this line in an else block corresponding to the if statement on line 48.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -28,24 +36,60 @@
...
+  private ArrayList<Task> tasks;
   @Override
-  public void init() {
-    trial_array = new ArrayList<>();
-    trial_array.add("First hard-coded message");
-    trial_array.add("Second hard-coded message");
-    trial_array.add("Third hard-coded message");
-    json_array = new Gson().toJson(trial_array);
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment-box");
```
Comment (mugdhalakhani) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Add some input sanitization here. What if the comment is empty?  Do you still want to post a comment?

File: portfolio/src/main/webapp/script.js [hunk 8]
```diff
@@ -32,15 +32,23 @@ function addRandomImage() {
...
+function createTaskElement(task) {
+  const taskElement = document.createElement('li');
+  taskElement.className = 'task';
+
+  const titleElement = document.createElement('span');
+  titleElement.innerText = task.comment + ' - Commented by ' + task.displayName;
+
+  taskElement.appendChild(titleElement);
```
Comment (mugdhalakhani) [inline portfolio/src/main/webapp/script.js:52]: Would this look nicer on a new line? (You can add a new line character \n here to add a newline)

**Review by mugdhalakhani — COMMENTED** [review]
One question about input sanitization, and a few nits. Nice work!

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 6 above)
Comment (mam249) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: will do in one of the later branches that are open

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 7 above)
Comment (mam249) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: have changed index.html to not accept no input but will look into if input is only spaces

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 9]
```diff
@@ -70,6 +73,11 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       tasks.add(task);
     
     }
+    Gson gson = new Gson();
+
+    response.setContentType("application/json;");
+    response.getWriter().println(gson.toJson(tasks));
+
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:80]: nit: remove newline before closing bracket

Comment (lminchillo) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:80]: (also in line 74)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 10]
```diff
@@ -56,10 +52,17 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     datastore.put(taskEntity);
     response.sendRedirect("/index.html");
-    
+  }
+  @Override
```
Comment (lminchillo) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:56]: newline before @Override

**Review by lminchillo — APPROVED** [review]
Nice work! Thanks for implementing the fixes/suggestions

**Pull request merged**